linked_list/implementation.py

```diff
@@ -1,40 +1,135 @@
 from .interface import AbstractLinkedList
-
+from linked_list.node import Node
 
 class LinkedList(AbstractLinkedList):
     """
     Implementation of an AbstractLinkedList inteface.
     """
-
     def __init__(self, elements=None):
-        pass
-
+        if not elements:
+            self.start = None
+            self.end = None
+        else:
+            new_node = Node(elements[0])
+            self.start = new_node
+            self.end = new_node
+            for element in elements[1:]:
+                self.append(element)
+        
     def __str__(self):
-        pass
+        l = []
+        iterNode = self.start
+        while iterNode is not None:
+            l.append(iterNode.elem)
+            iterNode = iterNode.next
+        return str(l)
+        
 
     def __len__(self):
-        pass
+        iterNode = self.start
+        count = 0
+        while iterNode is not None:
+            count += 1
+            iterNode = iterNode.next
+        return count
 
+# http://stackoverflow.com/questions/2180578/python-why-cant-i-iterate-over-a-list-is-my-exception-class-borked
     def __iter__(self):
```

Comment:
Pretty cool that you got the __iter__ method working. I even looked at the same stackoverflow question and didn't figure out the appropriate implementation and ended up going with a generator.